library/src/main/java/com/airbnb/android/airmapview/WebAirMapViewBuilder.java

```diff
@@ -13,13 +17,42 @@
   }
 
   /**
+   * Set Mapbox Access Token for Web map
+   * @link https://www.mapbox.com/help/define-access-token/
+   * @param mapboxAccessToken Mapbox Access Token
+   * @return WebAirMapViewBuilder
+   */
+  public WebAirMapViewBuilder setMapboxAccessToken(String mapboxAccessToken) {
+    this.mapboxAccessToken = mapboxAccessToken;
+    return this;
+  }
+
+  /**
+   * Set Mapbox Map Id
+   * @link https://www.mapbox.com/help/define-map-id/
+   * @param mapboxMapId
+   * @return WebAirMapViewBuilder
+   */
+  public WebAirMapViewBuilder setMapboxMapId(String mapboxMapId) {
+    this.mapboxMapId = mapboxMapId;
+    return this;
+  }
+
+  /**
    * Build the map fragment with the requested options
    *
    * @return The {@link WebViewMapFragment} map fragment.
    */
   @Override public WebViewMapFragment build() {
     if (options == null) {
-      options = new GoogleWebMapType();
+      if (!TextUtils.isEmpty(mapboxAccessToken) && !TextUtils.isEmpty(mapboxMapId)) {
+        options = new MapboxWebMapType(mapboxAccessToken, mapboxMapId);
+      } else {
+        options = new GoogleWebMapType();
+      }
+    }
+    if (options instanceof MapboxWebMapType) {
+      return MapboxWebViewMapFragment.newInstance(options);
```

Comment:
once we move all Mapbox logic to `MapboxWebViewMapFragment`, you can revert all your changes to this class